### PR TITLE
add esp32-s3-korvo-1 audio and sd card (BSP-446)

### PR DESCRIPTION
**Pull request by pborcin**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

### CODE REVIEW

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/idf_component.yml
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/idf_component.yml:20]: Let's allow automatic bugfix updates
```suggestion
    version: "~1.0.3"
```

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf5.c
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf5.c:61]: This TDM configuration is not usual. Could you provide a docs block explaining the settings?

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf4.c
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1_idf4.c:7]: This file was probably not updated and will fail to compile.

**Review by tore-espressif — COMMENTED** [review]
Thank you very much for the update @pborcin , I'd like to test on HW before we proceed with this MR.

2 more things:
1. This BSP is not build in any CI job. You should add `sdkconfig.bsp.esp32_s3_korvo_1` file to examples that should work on this BSP. Eg examples/audio
2. Audio for v4.4 will fail to build. Could you please have a look at that too?

**Review by espzav — COMMENTED** [review]
@pborcin Thank you for this PR. Good job! I didn't test it, I haven't got a board. Code LGTM Left only one comment and you should **increase the version of component**.

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:105]: Nitpick: Typo?
```suggestion
 * For speaker initialization use bsp_audio_codec_speaker_init() which in turn initializes I2S with bsp_audio_init().
```

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:26]: Could you also please add capabilities defines? Like this: https://github.com/espressif/esp-bsp/blob/master/bsp/esp-box-3/include/bsp/esp-box-3.h#L30

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:145]: @espzav It seems that when we added `bsp_audio_get_codec_itf` function to BSP APIs, we did not consider the scenarion where 2 I2S peripherals are used for playback and recording. What do you think about this new function?

Comment (tore-espressif) [reply bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:145]: Same applies to `bsp_audio_init` where we pass I2S config only for 1 I2S peripheral 😮‍💨

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (tore-espressif) [reply bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:35]: Thanks, could you add _all_ the capabilities macros? And define them as '0' in they are not supported? https://github.com/espressif/esp-bsp/blob/master/bsp/esp32_s3_usb_otg/include/bsp/esp32_s3_usb_otg.h#L25

**Comment by tore-espressif** [comment]
@pborcin I tried on HW and my audio playback is not working (no sound). Is there anything else I need to do to make it work?

**Review by tore-espressif — APPROVED** [review]
LGTM! @espzav should we wait for your review?

**Review by espzav — APPROVED** [review]
LGTM

File: bsp/esp32_s3_korvo_1/idf_component.yml
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/idf_component.yml:13]: @pborcin Please update version (line 1) to 1.1.0

**Review by tore-espressif — CHANGES_REQUESTED** [review]
@pborcin plese bump the version before merging!

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**
